### ISSUE TITLE
trailing non-whitespace characters are not considered errors

### ISSUE THREAD

**Issue by japaric** [dtolnay/miniserde#9]
### STR

``` rust
// serde_json = "=1.0.38"
// miniserde = "=0.1.9"

fn main() {
    println!("{:?}", miniserde::json::from_str::<bool>(" true && false "));
    println!("{:?}", serde_json::from_str::<bool>(" true && false "));
}
```

``` console
$ cargo +stable run
Ok(true)
Err(Error("trailing characters", line: 1, column: 7))
```

This behavior differs from `serde_json` but I didn't see it documented so I assume it's not intended behavior.

(Very cool library by the way)

**Comment by dtolnay** [comment]
Thanks! This is a bug. I would accept a PR to return an error on trailing non-whitespace.

**Comment by dtolnay** [comment]
Fixed in 0.1.10.
